tutorials/vbpr_text.ipynb

```diff
@@ -0,0 +1,215 @@
+    "\n",
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# install Cornac\n",
+    "!pip3 install numpy cornac\n",
+    "# install PyTorch for VBPR model\n",
+    "!pip3 install torch==1.0.0\n",
+    "\n",
+    "import cornac\n",
+    "from cornac.data import Reader\n",
+    "from cornac.datasets import movielens\n",
+    "from cornac.eval_methods import RatioSplit\n",
+    "from cornac.data import TextModality, ImageModality\n",
+    "from cornac.data.text import BaseTokenizer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prepare data\n",
+    "Here we use the MovieLens 100K dataset which is already accessible from Cornac. Hence, we can simply load movie plots and the rating data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plots, movie_ids = movielens.load_plot()\n",
+    "\n",
+    "# movies without plots are filtered out by `cornac.data.Reader`\n",
+    "ml_100k = movielens.load_100k(reader=Reader(item_set=movie_ids)) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Cross modality\n",
+    "\n",
+    "To get representations from text data, we build a `TextModality` using text corpus with coressponding ids as the input. We also need to supply a `Tokenizer` for text splitting, in this case the tokens in text plots are seperated by `\\tab` character. We limit the maximum size of vocabulary to 5000, which also means the size of representations (e.g., bag-of-words vectors) can not go beyond that limit."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "item_text_modality = TextModality(corpus=plots, ids=movie_ids, \n",
+    "                                  tokenizer=BaseTokenizer(sep='\\t', stop_words='english'),\n",
+    "                                  max_vocab=5000, max_doc_freq=0.5).build()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After building the `TextModality`, we can access text representations through provided APIs and attributes of the object. In this case, we take the word-count matrix as item representations then use it as input to construct an `ImageModality`. In other words, item visual representations will be substituted by bag-of-words representations which are used by `VBPR` model. "
```

Comment:
This part could be made more concise as follows: "The next step is to instantiate an `ImageModality` (which VBPR relies on) using our text representations, i.e., the word-count matrix in this case.  In other words, we will substitute text representations for visual features."

Reply:
noted